### PR TITLE
Miscellaneus latent worker test improvements

### PR DESCRIPTION
**Pull request by p12tic**
I've cherry-picked a couple of commits from #4616 that will need to land regardless of what form the PR will have in the end.

### CODE REVIEW

**Comment by tardyp** [comment]
Thanks for making it easier to review :)

File: master/buildbot/test/fake/latent.py [hunk 1]
```diff
@@ -156,7 +163,7 @@ def patchBot(self, case, remoteMethod, patch):
         case.patch(BotBase, remoteMethod, patch)
 
 
-class ControllableLatentWorker(AbstractLatentWorker):
+class ControllableLatentWorkerMixin(object):
```
Comment (rjarry) [inline master/buildbot/test/fake/latent.py]: Inheriting from `object` is not required anymore in python 3. Can you remove this?

File: master/buildbot/test/fake/latent.py [hunk 2]
```diff
@@ -165,19 +172,8 @@ class ControllableLatentWorker(AbstractLatentWorker):
...
-            build_wait_timeout=self._controller.build_wait_timeout,
-            **kwargs)
-
-    def reconfigService(self, name, _, **kwargs):
-        AbstractLatentWorker.reconfigService(
-            self, name, None,
-            build_wait_timeout=self._controller.build_wait_timeout,
-            **kwargs)
+        super(ControllableLatentWorkerMixin, self).__init__(
+            name, None, **kwargs)
```
Comment (rjarry) [inline master/buildbot/test/fake/latent.py]: Can you use the python 3 version of `super()` here?

```python
          super().__init__(name, None, **kwargs)
```

That way, it does not matter if this PR or #4620 is merged first.

File: master/buildbot/test/fake/latent.py [hunk 3]
```diff
@@ -216,3 +212,18 @@ def stop_instance(self, build):
...
+        AbstractLatentWorker.checkConfig(
+            self, name, None,
+            build_wait_timeout=self._controller.build_wait_timeout,
+            **kwargs)
+
+    def reconfigService(self, name, _, **kwargs):
+        AbstractLatentWorker.reconfigService(
+            self, name, None,
+            build_wait_timeout=self._controller.build_wait_timeout,
+            **kwargs)
```
Comment (rjarry) [inline master/buildbot/test/fake/latent.py]: same remark about super:

```python
super().checkConfig(name, None, build_wait_timeout=self._controller.build_wait_timeout, **kwargs)
```

and 

```python
super().reconfigService(name, None, build_wait_timeout=self._controller.build_wait_timeout, **kwargs)
```

File: master/buildbot/test/fake/latent.py (hunk 1 above)
Comment (p12tic) [reply master/buildbot/test/fake/latent.py]: Whoops, the commit is from before we dropped python 2. Fixed now.

File: master/buildbot/test/fake/latent.py (hunk 3 above)
Comment (p12tic) [reply master/buildbot/test/fake/latent.py]: Whoops, the commit is from before we dropped python 2. Fixed now.

File: master/buildbot/test/fake/latent.py (hunk 2 above)
Comment (p12tic) [reply master/buildbot/test/fake/latent.py]: Fixed now.

**Pull request merged**